File: data/multilang_comparison/program_cpp_vector/src/func18.cpp
#include "program_cpp_vector.hpp" 
Array* func18(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array125 = func19(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array125->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array125->refC--;
      if(array125->refC == 0) {
         array125->data.clear();
         delete array125;
         DEBUG_FREE(array125->id);
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array135 = func20(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array135->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array135->refC--;
      if(array135->refC == 0) {
         array135->data.clear();
         delete array135;
         DEBUG_FREE(array135->id);
      }
   }
   Array_param params0;
   params0.size = 0;
   std::vector<Array*> data_params0(0);
   params0.data = data_params0;
   Array* array140 = func21(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array140->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   return array140;
}

File: data/multilang_comparison/program_go/program_go/src/func18.go
package main
import "math/rand"
var _ = rand.Intn
func Func18(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array157 *Array = Func19(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array157.RefC--
      if array157.RefC == 0 {
          array157.Data = nil
          array157 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array167 *Array = Func20(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array167.RefC--
      if array167.RefC == 0 {
          array167.Data = nil
          array167 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array172 *Array = Func21(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array172;
}

